spec/models/article_spec.rb

```diff
@@ -1386,16 +1386,22 @@ def foo():
     end
   end
 
-  xit "does not send moderator notifications when a draft post" do
+  # This test has a history of being flaky, curreent theory is that the
+  # after_create_commit callback is not reliable
+  it "does not send moderator notifications when a draft post" do
+    # Another flakiness theory is that the class expectations aren't fully
+    # reset in some other test, so try to clear them out before setting ours:
+    RSpec::Mocks.space.proxy_for(Notification).reset
```

Comment:
I haven't been able to reproduce the flakiness myself, but when you did, were you able to confirm that `Notification.send_moderator_notification` was being called with this specific draft post?

Asking because maybe simply narrowing down the mock (i.e. `expect(Notification).not_to have_received(:send_moderation_notification).with(draft_post)` and `expect(Notification).to have_received(:send_moderation_notification).with(published_post)`) might work?